src/app/cli/src/commands/command_helpers.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Arc;
+
+use dill::{component, interface};
+use kamu::domain::utils::metadata_chain_comparator::CompareChainsResult;
+use kamu::domain::PushStatusService;
+use opendatafabric::DatasetHandle;
+
+use crate::{CLIError, Interact};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait CommandHelper: Send + Sync {
+    async fn confirm_deletion(&self, dataset_handles: &[DatasetHandle]) -> Result<(), CLIError>;
+}
```

Comment:
What is the point of separating this helper from the delete command?

To me types like "Utils" and "Helper" are always a sign of code smell. If you cannot give some component a clear name - it usually indicates that it's not logically decomposed correctly and you don't actually know what purpose it serves.

Also a good sanity test is to ask "what happens if all other commands need helpers?". Do we end up dumping code from multiple commands into this type? Or will we have `CommandHelper2` and `CommandHelper3`. This decomposition doesn't make sense to me.

And why do we need the complexity of hiding helper behind an interface and injecting via DI? Is it ever going to be replaced by another implementation?

Reply:
In both `add` and `delete` command we need to print sync warning and show delete confirmation. This code was moved to `confirm_deletion` function. As far as I didn't found better place, I decided to create `command_helpers` file in order to have such random helpers into that file. What would you propose here?

> And why do we need the complexity of hiding helper behind an interface and injecting via DI? Is it ever going to be replaced by another implementation?

interface is redundant, but I think it makes sense to keep it in DI, because it has 2 deps which we'll need to pass as function params otherwise.

Reply:
Ah, I missed that `confirm_deletion` is also used by Add command.

Would composition work better here? Can Add command instantiate and call the Delete command for those datasets that already exist when `--replace` flag is present, let Delete do its thing, and then proceed if Delete was successful?

Reply:
A command calling another command is an example of even worse decomposition.

Reply:
I clearly see this as a CLI-level service. Perhaps the name `CommandHelper` isn't brilliant, we could have `ConfirmDeleteService` at CLI level. It really does not need to have a trait and can be injected in commands directly, similarly to `WorkspaceService`.

Reply:
Or it could be a `CliInteractionService`, `CliPromptService`, or whatever similar.

Reply:
Switched back to service approach (with no trait)